### PR TITLE
Beginning of the Italian Rework

### PR DESCRIPTION
**Pull request by Eeillios**
All tested and working. Reviewed Monarchy Referendum and Italian Paris Peace Treaties. Added Umag province with no issues and removed 1945 events, exagerated events (graziani and badoglio's trials which either gave + or - 20% scialist and communist) and PCI event chain as it will not be needed anymore.

### CODE REVIEW

File: CWE/events/italy.txt [hunk 1]
```diff
@@ -44,6 +79,12 @@ country_event = {
 	option = {
 		name = EVT_8007704_B
 		define_general = { name = "Umberto II" personality = energetic background = no_background }
+		set_country_flag = umberto_II
+		any_pop = { 
+			    limit = { OR = { ideology = communist } OR = { ideology = socialist } }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:84]: "OR = { ideology = communist } OR = { ideology = socialist }" is wrong, it should be "OR = { pop_majority_ideology = communist pop_majority_ideology = communist_social }". 

It is better to use scaled militancy and consciousness for these kinds of effects.
any_pop = {
		scaled_militancy = {
			factor = 5
			ideology = communist
		}
		scaled_militancy = {
			factor = 5
			ideology = communist_social
		}
		scaled_consciousness = {
				ideology = communist
				factor = 6
		}
		scaled_consciousness = {
				ideology = communist_social
				factor = 6
		}
}

File: CWE/events/italy.txt [hunk 2]
```diff
@@ -1,24 +1,59 @@
...
+			}
+		any_pop = { 
+				ideology = { factor = 0.18 value = communist }
+			}
+		any_pop = { 
+			    limit = { OR = { ideology = communist } OR = { ideology = socialist } }
+				consciousness = 5
+				militancy = -5
+			}
+		set_country_flag = republic
```
Comment (settintotrieste) [inline CWE/events/italy.txt:56]: Please do not use country flags with just one generic word. Please use set_country_flag = ITA_1948_republic_flag instead

File: CWE/events/italy.txt [hunk 3]
```diff
@@ -1,24 +1,59 @@
...
+				ideology = { factor = 0.15 value = big_tent }
+			}
+		any_pop = { 
+				ideology = { factor = 0.20 value = socialist }
+			}
+		any_pop = { 
+				ideology = { factor = 0.18 value = communist }
+			}
+		any_pop = { 
+			    limit = { OR = { ideology = communist } OR = { ideology = socialist } }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:52]: "OR = { ideology = communist } OR = { ideology = socialist }" is wrong

File: CWE/events/italy.txt [hunk 4]
```diff
@@ -1,24 +1,59 @@
...
+	
+	fire_only_once = yes
+	
 	trigger = {
 		tag = ITA
-		year = 1946 NOT = { year = 1950 }
 		government = hms_government
-		war = no
-		is_vassal = no
+		NOT = { year = 1950 }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:13]: Please follow the style convention.

trigger = {
TAG
YEAR

GOV TYPE
is_vassal = yes/no
war = yes/no

ruling_party_ideology (if necessary)

country flags (if necessary)
}

File: CWE/events/italy.txt [hunk 5]
```diff
@@ -54,86 +95,6 @@ country_event = {
 	}
 }
 
-#ICP foundation
```
Comment (settintotrieste) [inline CWE/events/italy.txt:57]: Please do not remove these events unless there is an immediate replacement for them.

File: CWE/events/italy.txt [hunk 6]
```diff
@@ -825,31 +464,6 @@ country_event = {
 
 }
 
-#Pavarotti dies
```
Comment (settintotrieste) [inline CWE/events/italy.txt:828]: Please do not remove content that already works unless there is a replacement event for it or the even is causing issues.

File: CWE/events/trieste.txt [hunk 7]
```diff
@@ -1,88 +1,3 @@
-#Establish the Free City of Trieste
```
Comment (settintotrieste) [inline CWE/events/trieste.txt:1]: These events are necessary.

**Review by settintotrieste — CHANGES_REQUESTED** [review]

File: CWE/events/Essential Events 8.txt [hunk 8]
```diff
@@ -7357,6 +7357,21 @@ country_event = {
 	option = {
 		name = "May we have lasting peace in Europe"
 		add_country_modifier = { name = ger_war_rep duration = 3650 }
+                770 = { remove_core = ITA }
```
Comment (settintotrieste) [inline CWE/events/Essential Events 8.txt:7360]: I prefer to put this as part of the earlier set of events as otherwise the checks from there will have to be moved to this event

**Review by settintotrieste — CHANGES_REQUESTED** [review]

File: CWE/events/Essential events 3.txt [hunk 9]
```diff
@@ -2087,116 +2087,6 @@ country_event = {
 	}
 }
 
-#Pietro Badoglio War Trials
```
Comment (settintotrieste) [inline CWE/events/Essential events 3.txt:2090]: Please do not remove these events as they are still worthwhile ones to include.

**Review by settintotrieste — CHANGES_REQUESTED** [review]

**Comment by settintotrieste** [comment]
Did you use the latest version of the map file? Just to check as there was a change last week to fix a stay pixel in Indonesia.

File: CWE/events/Essential Events 8.txt (hunk 8 above)
Comment (settintotrieste) [reply CWE/events/Essential Events 8.txt:7360]: The better approach to link the peace treaty event and the formation of the free territory of Trieste event is to make the peace treaty even set a global flag which is checked for by the Trieste formation event.

File: CWE/events/italy.txt (hunk 4 above)
Comment (Eeillios) [reply CWE/events/italy.txt:13]: Missed the TAG whoops, but I realized that  without specifying the year id it's 1946 it fires more on time, while if I specify year = 1946 it maye fire inbetween may and january of the year after, tho if necessary will fix.

File: CWE/events/italy.txt (hunk 5 above)
Comment (Eeillios) [reply CWE/events/italy.txt:57]: I will be adding a first stage of that later today and if I can't today by the end of the week, the Foundation event is supposed to happen before gamestart and the rest of the chain doesn't really fit what will be added nor will work with it

File: CWE/events/italy.txt (hunk 6 above)
Comment (Eeillios) [reply CWE/events/italy.txt:828]: I prefered to re-purpose the event as the Pavarotti death seems kinda unfitting with the whole mod as no nation has events for pop culture deaths and ha sno relevance to gameplay nor any effect

File: CWE/events/trieste.txt (hunk 7 above)
Comment (Eeillios) [reply CWE/events/trieste.txt:1]: If they are necessary I will have to tie them to the Paris Peace Treaty and have them reviewed as OTL it had been made clear since before the end of the war that the territory was to be established and was a clause of the treaty

File: CWE/events/Essential events 3.txt (hunk 9 above)
Comment (Eeillios) [reply CWE/events/Essential events 3.txt:2090]: I will change the effects then since as of now they can break gameplay either by making communism too strong or too weak

**Comment by Eeillios** [comment]
I used the latest version yes

File: CWE/events/trieste.txt (hunk 7 above)
Comment (settintotrieste) [reply CWE/events/trieste.txt:1]: Okay but please tie them by making the Paris peace treaty event set a global flag for this event to fire instead of deleting these events.

File: CWE/events/Essential events 3.txt (hunk 9 above)
Comment (settintotrieste) [reply CWE/events/Essential events 3.txt:2090]: Thanks

File: CWE/events/italy.txt (hunk 6 above)
Comment (settintotrieste) [reply CWE/events/italy.txt:828]: Okay

File: CWE/events/italy.txt (hunk 5 above)
Comment (settintotrieste) [reply CWE/events/italy.txt:57]: Alright

File: CWE/events/italy.txt (hunk 4 above)
Comment (settintotrieste) [reply CWE/events/italy.txt:13]: I feel it is better to specify a date to make the code clearer for third parties who may want to edit these events in the future.

File: CWE/events/trieste.txt [hunk 10]
```diff
@@ -78,6 +80,50 @@ country_event = {
...
+	}
+  
+	option = {
+		name = "Yugoslavia ought to respect the treaties" 
+		badboy = 5
+		relation = { who = YUG value = -100 } 
+		relation = { who = ITA value = 50 }
+		war = { 
+			target = YUG
+			attacker_goal = { casus_belli = free_allied_cores }
```
Comment (settintotrieste) [inline CWE/events/trieste.txt:114]: Did you test whether free allied cores works for this event? Iirc this CB sometimes doesn’t work for scripted events.

File: CWE/events/italy.txt [hunk 11]
```diff
@@ -54,460 +99,232 @@ country_event = {
...
-			defender_goal = { casus_belli = humiliate } 
-		}
-		
-		#USSR choice to defend
-		random_country = {
-			limit = { tag = RUS government = proletarian_dictatorship }
-			country_event = 8007745 #RUS choice to ally with ITA
-		}
+		name = "Enact the new laws!"
+		relation = { who = USA value = -100 }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:304]: Please use a random_country scope to affect the relations, following the example below.

random_country = {
 			limit = { tag = USA government = democracy is_vassal = no }
 			relation = { who = THIS value = -100 }
 		}

File: CWE/events/Essential events 3.txt [hunk 12]
```diff
@@ -2087,18 +2087,20 @@ country_event = {
...
+-#Pietro Badoglio War Trials
```
Comment (settintotrieste) [inline CWE/events/Essential events 3.txt:2090]: A stray hyphen was added.

File: CWE/events/italy.txt [hunk 13]
```diff
@@ -54,460 +99,232 @@ country_event = {
...
+		is_vassal = no
+		war = no
+		
+		has_country_flag = ITA_1948_republic_flag
 	}
 
-	mean_time_to_happen = { months = 2 }
+	mean_time_to_happen = { months = 1 }
 
 	option = {
```
Comment (settintotrieste) [inline CWE/events/italy.txt:163]: Please provide an option to not adopt the 1948 constitution.

File: CWE/events/italy.txt [hunk 14]
```diff
@@ -54,460 +99,232 @@ country_event = {
...
-	option = {
-		name = EVT_8007735_B
-		ai_chance = { factor = 0.2 }
-		any_pop = { militancy = 3 }
-		any_pop = {
-			limit = { is_primary_culture = yes }
-			ideology = { factor = 0.05 value = conservative }
-		}
-		set_country_flag = very_radical_ICP 
+		ai_chance = { factor = 1 }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:178]: Please do not have any historical events with factor = 1. All historical event must have at least some percentage chance for an ahistorical option.

File: CWE/events/italy.txt [hunk 15]
```diff
@@ -54,460 +99,232 @@ country_event = {
...
-		any_pop = {
-			limit = { is_primary_culture = yes }
-			ideology = { factor = 0.05 value = socialist }
+		name = "Centrist Coalition (DC-US-BN-PRI)"
+		ruling_party_ideology = conservative
+		any_pop = { 
+			ideology = { factor = 0.1 value = conservative }
 		}
+		set_country_flag = 1948_ITA_Centrism
 		ai_chance = { factor = 0.7 }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:211]: If there are 2 options in a historical event, the minimum chance for the historical option must be 90%.

File: CWE/events/italy.txt [hunk 16]
```diff
@@ -54,460 +99,232 @@ country_event = {
...
+		NOT = { year = 1990 }
 		
-		#Citizens become more communist
-		any_pop = {
-			limit = { is_primary_culture = yes }
-			ideology = { factor = 0.05 value = communist }
-		}
+		government = democracy
+		is_vassal = no
+		war = no
```
Comment (settintotrieste) [inline CWE/events/italy.txt:295]: There should be a check for a communist or socialist ruling party here since this event is to create a proletarian dictatorship

**Review by settintotrieste — CHANGES_REQUESTED** [review]

File: CWE/events/trieste.txt (hunk 10 above)
Comment (Eeillios) [reply CWE/events/trieste.txt:114]: I haven't reached that part of testing yet as I didn't know that if I committed to my branch you would see it and be already part of the pull. As of now no event after the referendum seems to fire and when I try to console them it says they don't exist and I'm searching the cause, so I will check that later

File: CWE/events/italy.txt (hunk 15 above)
Comment (Eeillios) [reply CWE/events/italy.txt:211]: wait so no 70/30 possibilities if it was a tight race OTL?

File: CWE/events/italy.txt (hunk 16 above)
Comment (Eeillios) [reply CWE/events/italy.txt:295]: I feared that with the vanilla elections system the ruling party in that period might not actually be communist or socialist and that would prevent the event from firing even if all the other prerequisites are met, if it is necessary I will try to tweak events boosts to secure a solid majority for when the event fires

File: CWE/events/italy.txt [hunk 17]
```diff
@@ -1,24 +1,63 @@
...
+			}
+		any_pop = { 
+				ideology = { factor = 0.20 value = socialist }
+			}
+		any_pop = { 
+				ideology = { factor = 0.18 value = communist }
+			}
+		any_pop = { 
+			    limit = { OR = { pop_majority_ideology = socialist pop_majority_ideology = communist pop_majority_ideology = communist_social } }
+				scaled_consciousness = 5
```
[the remark this answers is not in the document]
Comment (Eeillios) [reply CWE/events/italy.txt:57]: weird, it was working as intended with that phrasing and now not anymore

File: CWE/events/italy.txt (hunk 15 above)
Comment (settintotrieste) [reply CWE/events/italy.txt:306]: Yeah, I like it at 90% so it’s easier for testing and also so people don’t keep complaining that ahistorical events keep happening and spoiling their game.

File: CWE/events/italy.txt (hunk 17 above)
[the remark this answers is not in the document]
Comment (settintotrieste) [reply CWE/events/italy.txt]: Scaled_militancy = 5 is wrong. If you want to affect pops of a certain ideology only then you have to use the code above.

File: CWE/events/italy.txt [hunk 18]
```diff
@@ -175,7 +256,21 @@ country_event = {
...
 
-		ai_chance = { factor = 1 }
+		ai_chance = { factor = 0.99 }
+	}
+	
+	option = {
+		name = "Strive for a different constitution!"
+		prestige = -15
+		
+		any_pop = {
```
Comment (settintotrieste) [inline CWE/events/italy.txt:266]: This is wrong. It should be 

any_pop = { consciousness = 2 militancy = 2 }

File: CWE/events/italy.txt [hunk 19]
```diff
@@ -301,10 +422,22 @@ country_event = {
...
-		relation = { who = USA value = -100 }
-		relation = { who = ENG value = -100 }
-		relation = { who = FRA value = -100 }
-		relation = { who = RUS value = 50 }
+		random_country = {
+                       limit = { tag = USA government = democracy is_vassal = no }
+                       relation = { who = THIS value = -100 }
+        }
+		random_country = {
+                       limit = { tag = ENG government = democracy is_vassal = no }
```
Comment (settintotrieste) [inline CWE/events/italy.txt:430]: The check for ENG should be government = hms_government since they are a constitutional monarchy

**Review by settintotrieste — CHANGES_REQUESTED** [review]

File: CWE/events/italy.txt [hunk 20]
```diff
@@ -54,460 +126,341 @@ country_event = {
...
+        any_pop = {
+                scaled_consciousness = { factor = 2 ideology = liberal }
+			}
+		ai_chance = { factor = 0.1 }
 	}
+   
 }
 
-#Togliatti assassination attempt
+#1948 Constitution
```
Comment (settintotrieste) [inline CWE/events/italy.txt:219]: Constitution event must be triggered immediately by the republic formation event because otherwise the generic event to choose republic type will fire in between. Alternative is to set Italy as a parliamentary republic in the ITA republic formation event first to block the generic event then this event just chooses whether to leave it as so or to pick another republic type, example presidential republic.

File: CWE/events/italy.txt [hunk 21]
```diff
@@ -54,460 +126,341 @@ country_event = {
...
+			
+		set_country_flag = parliamentary_system_republic
 
+		ai_chance = { factor = 0.99 }
+	}
+	
 	option = {
-		name = EVT_8007736_C
-		any_pop = { militancy = 10 }
+		name = "Strive for a different constitution!"
```
Comment (settintotrieste) [inline CWE/events/italy.txt:263]: This option should select another republic type, possibly presidential so as to block the generic event from firing when then allows parliamentary to be selected again.

File: CWE/events/italy.txt [hunk 22]
```diff
@@ -54,460 +126,341 @@ country_event = {
...
+  desc = EVT_8007736_DESC
+  picture = "nwo2_palmiro_togliatti"
   fire_only_once = yes
 
 	trigger = {
 		tag = ITA
-		year = 1960 NOT = { year = 1962 }
-		has_country_flag  = very_moderate_ICP
-		OR = { government = democracy government = hms_government }
-		NOT = {
```
Comment (settintotrieste) [inline CWE/events/italy.txt:264]: The ruling party check should be retained since if Togliatti was PM then he would have better security, preventing the assassination event.

File: CWE/events/italy.txt [hunk 23]
```diff
@@ -54,460 +126,341 @@ country_event = {
...
-		has_global_flag = blocsenabled
-		ruling_party_ideology = communist
-		government = proletarian_dictatorship
-		election = no
-	}
-
-	option = {
-		name = EVT_8007743_A
-		
-		#Break alliances with non communists
```
Comment (settintotrieste) [inline CWE/events/italy.txt:413]: Break alliances event must be retained since a communist state is being formed.

File: CWE/events/italy.txt [hunk 24]
```diff
@@ -54,460 +126,341 @@ country_event = {
...
-		any_country = {
-			limit = { NOT = { ruling_party_ideology = communist } }
-			leave_alliance = THIS
-			relation = { who = THIS value = -200 }
-		}
-		
-		#Be friends with fellow communists
```
Comment (settintotrieste) [inline CWE/events/italy.txt:420]: This should also be retained since fellow communists would be pleased with this development

File: CWE/events/italy.txt [hunk 25]
```diff
@@ -54,460 +126,341 @@ country_event = {
...
-#RUS choice to defend Communist Italy
-country_event = {
-  id = 8007745
-  title = EVT_8007745_NAME
-  desc = EVT_8007745_DESC
-  picture = "nwo2_italian_communist_party"
-  is_triggered_only = yes # RUS
+		government = proletarian_dictatorship
+      
+		political_reform = regulated_opinion
```
Comment (settintotrieste) [inline CWE/events/italy.txt:444]: It might be better to use country_event = 800052 to set the reforms when setting up a communist state.

**Review by settintotrieste — CHANGES_REQUESTED** [review]

File: CWE/events/italy.txt (hunk 16 above)
Comment (settintotrieste) [reply CWE/events/italy.txt:416]: It’s better to make sure a socialist or communist party is in power for that event to fire.

File: CWE/events/italy.txt (hunk 22 above)
Comment (Eeillios) [reply CWE/events/italy.txt:264]: The shooting is kinda needed to justify more restrictive laws and whetevr we can use reality to base AltHis it's better, also it wouldn't be that sure he would get better security, many even in positions of power would very much not like him and he had made the Church an enemy which is a big threat in italian politics

**Comment by Eeillios** [comment]
Thank you so much for the help, sorry for all the troubles for the next events I will be much more careful

**Comment by settintotrieste** [comment]
No problem and thank you very much for your additions

File: CWE/events/italy.txt [hunk 26]
```diff
@@ -31,39 +31,47 @@ country_event = {
...
+	
+	is_triggered_only = yes
 
 	option = {
 		name = EVT_8007704_A
+		
 		government = democracy
+		
 		any_pop = { 
-				ideology = { factor = 0.20 value = conservative }
```
Comment (Eeillios) [inline CWE/events/italy.txt]: why did you remove the boost for all ideologies? I put it there to make it so the pops have the same ideologies as the ones from the 1946 constituent assembly elections and so that communists have enough of a support base to represent OTL PCI popularity

File: CWE/decisions/italy.txt [hunk 27]
```diff
@@ -0,0 +1,95 @@
+political_decisions = {
+
+	ina_casa = {
+		picture = "piano_fanfani"
+		potential = {
+		    year = 1949
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:6]: Please follow the coding convention of:

TAG
Year
Government 
Vassal yes or no
War 
Ruling party
Money
Provinces owned
Flags

File: CWE/decisions/italy.txt [hunk 28]
```diff
@@ -0,0 +1,95 @@
...
+		}
+		allow = {
+			OR = { ruling_party_ideology = conservative ruling_party_ideology = big_tent }
+			money = 200000
+			is_vassal = no
+		}
+		effect = {
+			prestige = 10
+			money = -100000
+			political_reform = social_abuse_protection
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:21]: Please put a check for social abuse protection or better to not have already been enacted.

File: CWE/decisions/italy.txt [hunk 29]
```diff
@@ -0,0 +1,95 @@
...
+		}
+		allow = {
+			OR = { ruling_party_ideology = conservative ruling_party_ideology = big_tent ruling_party_ideology = socialist ruling_party_ideology = communist }
+			money = 400000
+			is_vassal = no
+		}
+		effect = {
+			prestige = 10
+			money = -200000
+			any_pop = { limit = { type = farmers } money = 30 }
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:47]: This would not work.

it has to be farmers = { money = 30 }

File: CWE/decisions/italy.txt [hunk 30]
```diff
@@ -0,0 +1,95 @@
...
+	
+	casmez = {
+		picture = "casmez"
+		potential = {
+		    year = 1950
+			tag = ITA
+			government = democracy
+			has_country_flag = ITA_1948_republic_flag
+			NOT = { has_country_flag = italy_dirigisme }
+			NOT = { has_country_flag = ITA_Casmez }
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:63]: There should be a trigger check for the relevant provinces to be owned.

File: CWE/decisions/italy.txt [hunk 31]
```diff
@@ -0,0 +1,95 @@
...
+		allow = {
+			OR = { ruling_party_ideology = conservative ruling_party_ideology = big_tent ruling_party_ideology = socialist ruling_party_ideology = communist }
+			money = 400000
+			is_vassal = no
+		}
+		effect = {
+			prestige = 10
+			money = -200000
+		PAP_752  = { 
+			any_pop = { OR = { type = capitalists type = bureaucrats } money = 100 }
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:74]: This would not work.

it has to be pop_type = { money = X }

File: CWE/decisions/italy.txt [hunk 32]
```diff
@@ -0,0 +1,95 @@
+political_decisions = {
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:1]: Please set the pre 1949 decisions as already fired in the history entry of Italy, under the 1950 bracket. See USA’s history file for an example.

File: CWE/decisions/italy.txt [hunk 33]
```diff
@@ -0,0 +1,95 @@
...
+		}
+		allow = {
+			OR = { ruling_party_ideology = conservative ruling_party_ideology = big_tent ruling_party_ideology = socialist ruling_party_ideology = communist }
+			money = 400000
+			is_vassal = no
+		}
+		effect = {
+			prestige = 10
+			money = -200000
+		PAP_752  = {
```
Comment (settintotrieste) [inline CWE/decisions/italy.txt:73]: I recommend using any_owned = { limit = { OR = { region = ALD_1714 } } … } instead

**Review by settintotrieste — CHANGES_REQUESTED** [review]

**Comment by settintotrieste** [comment]
Re: why did you remove the boost for all ideologies? I put it there to make it so the pops have the same ideologies as the ones from the 1946 constituent assembly elections and so that communists have enough of a support base to represent OTL PCI popularity

I felt it wouldn’t make much sense for the rich strata to support socialists and communists. I don’t think it would affect the ability of socialists or communists to win elections as percentage wise the rich strata is very small. Also, it helps create tension between the classes which would make the scaled militancy effects more consequential to the player.

**Comment by Eeillios** [comment]
Yeah I actually realized that you just made it limited to strata later, so I deleted the comment but it reached you anyways. If it's alright for you once the decisions are fixed you could merge the pull request as content for the 40's is all done.

**Comment by settintotrieste** [comment]
Thanks, I’ll do a small code cleanup then merge this PR.
